### PR TITLE
Upgrade mesh support

### PR DESCRIPTION
**Pull request by kitchoi**
Hi @dpinte  and @itziakos,

I have added support for `vtk_mesh`.  Tests and examples related to mesh are also edited.

Other changes _not_ related to `vtk_mesh`:
- `TestLatticeSource`: I overlooked it in the upgrade-lattice-support PR.
- test_cuba_data_extractor.py : changes related to `Particle`

Failed tests and test errors are related to `Particle` and `CUDS_file_source`, which will be dealt with in a separate branch later.

### CODE REVIEW

File: simphony_mayavi/cuds/vtk_mesh.py [hunk 1]
```diff
@@ -215,15 +241,15 @@ def get_point(self, uid):
...
-            raise ValueError(message.format(point.uid))
-        # Need to cast to int https://github.com/enthought/mayavi/issues/173
-        self.data_set.points[int(index)] = point.coordinates
-        self.point_data[index] = point.data
+    def update_points(self, points):
+        for point in points:
+            try:
+                index = self.point2index[point.uid]
+            except KeyError:
+                message = "Point with {} does exist"
```
Comment (dpinte) [inline simphony_mayavi/cuds/vtk_mesh.py]: does `not` exist ?

File: simphony_mayavi/cuds/vtk_mesh.py [hunk 2]
```diff
@@ -235,15 +261,16 @@ def iter_points(self, uids=None):
...
-            raise ValueError(message.format(type(element), element.uid))
-        point_ids = [self.point2index[uid] for uid in element.points]
-        self.elements[index] = point_ids
-        self.element_data[index] = element.data
+    def _update_elements(self, elements):
+        for element in elements:
+            try:
+                index = self.element2index[element.uid]
+            except KeyError:
+                message = "{} with {} does exist"
```
Comment (dpinte) [inline simphony_mayavi/cuds/vtk_mesh.py]: does `not` exist?

File: simphony_mayavi/cuds/vtk_mesh.py (hunk 1 above)
Comment (kitchoi) [inline simphony_mayavi/cuds/vtk_mesh.py]: Aha!  Yes you're right!

**Comment by dpinte** [comment]
@kitchoi I made some minor comments. Ideally you would separate the non vtk-mesh related commits in separate PR's. For the rest, this looks good to merge for me.

**Pull request merged**